raft/src/main/java/org/apache/kafka/raft/internals/BatchAccumulator.java

```diff
@@ -147,6 +109,13 @@ private long append(int epoch, List<T> records, boolean isAtomic) {
 
         appendLock.lock();
         try {
+            long endOffset = nextOffset + records.size() - 1;
+            requiredEndOffset.ifPresent(r -> {
```

Comment:
If the user of `RaftClient` provides the "expected base offset" this becomes `expectedBaseOffset.ifPresent(baseOffset -> if (baseOffset == nextOffset) { ... } );`